core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import reactor.core.Disposable;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This class offer API that simplifies the task of executing long-running operations against Azure. The poller consists of
+ * {@link Function} to be triggered on each poll cycle followed by polling to determine if long-running operation completed.
+ * <p>
+  * It provides following API:
+ *
+ * <ul>
+ *      <li>Querying the current state of long-running operations.</li>
+ *      <li>Requesting an asynchronous notification for long-running operations's state.</li>
+ *      <li>Cancelling the long-running operations if cancellation is supported by the service.</li>
+ *      <li>Triggering a poll operation manually.</li>
+ *      <li>Enable/Disable auto-polling.</li>
+ * </ul>
+ *
+ * <p><strong>Auto Polling</strong></p>
+ * Auto-polling is enabled by-default. It means that the {@link Poller} starts polling as soon as it's instance is created. The {@code Poller} will call poll operation every polling cycle
+ * and maintain state of long-running operation. Azure service can return {@code PollResponse#retryAfter} to override {@code pollInterval} defined in {@code Poller}.
+ * The {@code Poller.status} represents the status returned by the successful long-running operation at the time the last auto-polling or last manual polling, whichever happened most recently.
+ *
+ *<p><strong>Disable Auto Polling</strong></p>
+ * Those scenario which requires manual control of poll cycle, you can disable auto-poling by calling {@code setAutoPollingEnabled#false}. You can perform manual poll
+ * by invoking {@code poll} function. It will perform one call to poll operation and updated {@code poller} with latest status.
```

Comment:
```suggestion
 * by invoking {@code poll} function. It will perform one call to poll operation and update the {@code poller} with the latest status.
```